### ISSUE TITLE
Tame Beast should convert target creature into pet rather than creating a new entity

### ISSUE THREAD

**Issue by Ovahlord** [TrinityCore/TrinityCore#22532]
**Description:**
After some digging into the pet system I just saw that the beast taming procedure is completely wrong. Checking sniffs have confirmed that once you tame a creature the creature itself becomes your pet by simply updating all fields.
Once this happened the pet casts  Spell ID: 24592 (Hunter Pet Buffs)   to apply all passives for the scalings and on 4.x and newer it also casts Spell ID: 99289 (Energize) to fill up the focus.

**Current behaviour:**
The core currently despawns the tame target and creates a new creature at its position.

**Expected behaviour:**
The tame target should not despawn but instead become the new pet directly by converting it including stats.

**Steps to reproduce the problem:**

1. Just tame a pet on retail and then tame one on TC.

**Branch(es):**
Both.

**TC rev. hash/commit:** 
9ec3724d34d5953fbff2653b1bdce411a6bac96e

**Operating system:**
 Win10 +64

**Comment by illfated** [comment]
Thanks. I felt something was incorrect about it, but I did not know exactly what. There is not supposed to be a "level up" animation when you tame a pet.

**Comment by Ovahlord** [comment]
Well, this actually can be fixed with the current system right now. There is a ugly hack that is setting the level -1 before returning it to normal again to artifically create this effect.

**Comment by nagrandbugtracker** [comment]
Sometimes it takes for hunters to abandon and re-take the quest because all three charges are used and creature is not tamed.

**Comment by illfated** [comment]
@nagrandbugtracker :
I have never seen that specific issue you mention during my 5 years on TC,
when doing the level 10 hunter quests to learn Tame Beast. Then again,
I only play on my own computer (which also runs auth+world+MySQL).

**Comment by Shauren** [comment]
Heh, good luck with this trying to convert Creature object into Pet

**Comment by jameyboor** [comment]
@Shauren `reinterpret_cast<Pet*>(m_creature)` now where's my award

**Comment by Shauren** [comment]
enjoy your crash tho

**Comment by jameyboor** [comment]
You should know I have top of the line state of the art crash detection that catches every seg fault and just ignores it.

**Comment by Shauren** [comment]
but does it keep people online?

**Comment by jameyboor** [comment]
If there were people on my server in the first place it would.. I'm depressed now

**Comment by Treeston** [comment]
For the sake of everyone's notifications, keep discussion on topic.

**Comment by illfated** [comment]
From my point of view it would be an improvement to stop seeing "level up" animation on successful taming when beast level == hunter level.
(Yeah, I know... cosmetic and not as important as things which actually don't work.)

**Comment by Killyana** [comment]
This spells are used for what? Creating the first pet or summoning?

    https://woehead.way-of-elendil.fr/?spell=46718 (Pet Boar)
    https://woehead.way-of-elendil.fr/?spell=46719 (Pet Carrion)
    https://woehead.way-of-elendil.fr/?spell=46720 (Pet Cat)
    https://woehead.way-of-elendil.fr/?spell=46721 (Pet Dragonhawk)
    https://woehead.way-of-elendil.fr/?spell=64330 (Pet Bear)
    https://woehead.way-of-elendil.fr/?spell=64331 (Pet Crab)
    https://woehead.way-of-elendil.fr/?spell=64332 (Pet Crocolisk)
    https://woehead.way-of-elendil.fr/?spell=64333 (Pet Gorilla)

**Comment by illfated** [comment]
Looking at the wowhead counterpart of those pages, it seems to be the summon spell for already tamed pets:
- https://www.wowhead.com/spell=46718/pet-boar
- https://www.wowhead.com/spell=46719/pet-carrion-bird
- https://www.wowhead.com/spell=46720/pet-cat
- https://www.wowhead.com/spell=46721/pet-dragonhawk
- https://www.wowhead.com/spell=64330/pet-bear
- https://www.wowhead.com/spell=64331/pet-crab
- https://www.wowhead.com/spell=64332/pet-crocolisk
- https://www.wowhead.com/spell=64333/pet-gorilla

Looking back at the difference between taming a beast for a permanent pet, compared to the quest spell, which should only modify the existing creature and return a wild/aggressive beast when dismissed, I am still very uncertain how Blizzard handles beast taming for permanent pets.

**Comment by offl** [comment]
Because of the id range, quite possible they're related to Arena Tournament

**Comment by illfated** [comment]
Fair point. I didn't think that much about the spell IDs at first, but now that you mention it, it looks plausible.

I presume we should look more at low ID spells like these:
- https://www.wowhead.com/spell=1515/tame-beast
- https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?spell=1579

**Comment by Killyana** [comment]
Arena Tournament?

**Comment by offl** [comment]
Yes, https://wow.gamepedia.com/Arena_Pass

**Comment by offl** [comment]
And gossip options for https://wow.gamepedia.com/Beastmaster_(arena)
```sql
-- tbc
SELECT * FROM `broadcast_text` WHERE `ID` BETWEEN 25327 AND 25344;
-- wotlk
SELECT * FROM `broadcast_text` WHERE `ID` BETWEEN 34067 AND 34083;
```
Now there's no doubt they related to ATR
